### ISSUE TITLE
Crash in Backward-Pass of grouped Conv1D https://github.com/tensorflow/tensorflow/issues/50397

### ISSUE THREAD

**Issue by ymodak** [keras-team/keras#14896]
System information

Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
TensorFlow installed from (source or binary): Binary
TensorFlow version (use command below): 2.5.0
Python version: 3.7
CUDA/cuDNN version: n/a
GPU model and memory: n/a
Backward-Pass of grouped Convolution (Conv1D) crashes:

InvalidArgumentError:  filter_size does not have enough elements, requested 896, got 224
	 [[node gradient_tape/sequential/conv1d/conv1d/Conv2DBackpropFilter (defined at <ipython-input-6-c99a994f0883>:1) ]] [Op:__inference_train_function_898]
Forward-Pass (simple call) works.

Code is available at Colab:
https://colab.research.google.com/drive/1yyxtQCeM81cEUAVA6_z1PdtEY2P5kZ2C?usp=sharing

cross-posted from TF Core : https://github.com/tensorflow/tensorflow/issues/50397
Reported by : @ChWick

**Comment by ashaw596** [comment]
I tested it. It works on GPU, breaks on CPU and TPU.

**Comment by amirgholami** [comment]
We also have encountered the same problem. Is there anyone looking into this? It seems no one is assigned to this bug.

**Comment by saadsaifse** [comment]
any update on this issue? My Mac doesn't have a GPU, so I am blocked until it gets resolved on CPU

**Comment by mattdangerw** [comment]
I believe this is a general issue with grouped convolutions not being supported on CPU (coupled with a not great error message). There is some more discussion here https://github.com/tensorflow/tensorflow/issues/51825

A workaround for now would be to force XLA compilation of the convolution layer. We are looking into whether we can do this workaround inside of keras.

```
# Use this instead of Conv1D to be able to do grouped convolutions on CPU.
class GroupedConv1D(tf.keras.layers.Conv1D):
  @tf.function(jit_compile=True)
  def call(self, inputs):
    return super().call(inputs)
```

**Comment by mattdangerw** [comment]
This should now be fixed on nightly and 2.8 (when it gets released).
